ldms/src/ldmsd/ldmsd_request.c

```diff
@@ -6680,3 +6699,131 @@ static int set_default_authz_handler(ldmsd_req_ctxt_t reqc)
 
 	return 0;
 }
+
+struct cmd_line_opts {
+	const char *l;
+};
+struct cmd_line_opts opts[] = {
+		['A'] = { "default_auth_args" },
+		['a'] = { "default_auth"   },
+		['B'] = { "banner"         },
+		['H'] = { "host_name"       },
+		['k'] = { "publish_kernel" },
+		['l'] = { "log_file"        },
+		['m'] = { "set_memory"            },
+		['n'] = { "daemon_name"    },
+		['P'] = { "worker_threads"    },
+		['r'] = { "pid_file"        },
+		['s'] = { "kernel_file"    },
+		['v'] = { "log_level"       },
+};
+
+extern int ldmsd_process_cmd_line_arg(char opt, char *value);
+static int cmd_line_arg_set_handler(ldmsd_req_ctxt_t reqc)
```

Comment:
@nichamon if we're going to have this option syntax, what do you think about converting the input string to an argv and an argc and then using getopt to parse it? This will remove all that strcmp code you have in there and allow you to use the getopts #define and getopt_long structure definition you already have.

Reply:
@tom I like the idea to provide the long options not only in configuration files but also at the command line. Converting a string to argc and argv should be trivial and doesn't need any complicated parsing, so I'll make the change.